### PR TITLE
Expose container view through API

### PR DESCRIPTION
**Pull request by cerihughes**
This PR:

- Simplifies the `HUBContainerView` API to just declare a "content view" as a `readonly` `UIScrollView`
- Exposes `HUBContainerView` through the public API via the `HUBViewController`
- Takes all `UIScrollView`-specific pass-through logic out of the VCs. Clients should directly modify the the view's contentView now.
- Internally, we use `HUBCollectionContainerView` as the `UICollectionView`-specific subclass of `HUBContainerView`. We can move more `UICollectionView`-specific code into this subclass in future PRs.

Note: This is an API breaking change.

### CODE REVIEW

**Comment by cerihughes** [comment]
@mhallendal @bubski @ynmehel @krris @poncel

@spotify/objc-dev

File: sources/HUBCollectionContainerView.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#import "HUBContainerView.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A collection view-based implementation of the container view.
+@interface HUBCollectionContainerView : HUBContainerView
+
+/**
+ *  Redefine the container view property to be a UICollectionView, and makde it readwrite for internal use.
```
Comment (krris) [inline sources/HUBCollectionContainerView.h]: 🔬 Typo: `make`

File: sources/HUBViewControllerImplementation.m [hunk 2]
```diff
@@ -165,7 +165,7 @@ - (void)dealloc
 
 - (void)loadView
 {
-    HUBContainerView *containerView = [[HUBContainerView alloc] initWithFrame:CGRectZero];
+    HUBCollectionContainerView *containerView = [[HUBCollectionContainerView alloc] initWithFrame:CGRectZero];
```
Comment (krris) [inline sources/HUBViewControllerImplementation.m]: What about adding a `const`? 😉

File: sources/HUBViewControllerExperimentalImplementation.m [hunk 3]
```diff
@@ -161,7 +160,7 @@ - (void)dealloc
 
 - (void)loadView
 {
-    HUBContainerView *containerView = [[HUBContainerView alloc] initWithFrame:CGRectZero];
+    HUBCollectionContainerView *containerView = [[HUBCollectionContainerView alloc] initWithFrame:CGRectZero];
```
Comment (krris) [inline sources/HUBViewControllerExperimentalImplementation.m]: What about adding a `const`? 😉

File: sources/HUBViewControllerImplementation.m (hunk 2 above)
Comment (cerihughes) [inline sources/HUBViewControllerImplementation.m]: Will do.

File: sources/HUBCollectionContainerView.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+#import "HUBContainerView.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A collection view-based implementation of the container view.
+@interface HUBCollectionContainerView : HUBContainerView
+
+/**
+ *  Redefine the content view property to be a UICollectionView, and makde it readwrite for internal use.
```
Comment (bubski) [inline sources/HUBCollectionContainerView.h]: Typo
>and makde it

File: include/HubFramework/HUBViewController.h [hunk 5]
```diff
@@ -174,6 +175,9 @@ NS_ASSUME_NONNULL_BEGIN
 /// The view controller's delegate. See `HUBViewControllerDelegate` for more information.
 @property (nonatomic, weak, nullable) id<HUBViewControllerDelegate> delegate;
 
+/// Redeclare the view property to be a HUBContainerView.
+@property(null_resettable, nonatomic, strong) HUBContainerView *view;
```
Comment (bubski) [inline include/HubFramework/HUBViewController.h]: Missing space after `@property`

**Pull request merged**